app/api/generateCharacterImage/route.ts

```diff
@@ -0,0 +1,56 @@
+import { kv } from '@vercel/kv';
+import axios from 'axios';
+import { GenerateCharacterImageRequest } from '@/lib/types';
+
+export async function POST(req: Request): Promise<Response> {
+    try {
+        const body = await req.json() as GenerateCharacterImageRequest;
+        if (typeof body !== 'object') {
+            throw new Error('Invalid request body: expecting object');
+        }
+        const characterDescription = body.characterDescription; // replace with actual data fields from your request
+        const openAIResponse = await callAzureOpenAI(characterDescription)
+        const imageUrl = openAIResponse.data[0].url; 
+
+        // Download the image 
+        const imageResponse = await axios.get(imageUrl, {responseType: 'arraybuffer'});
+        const imageBuffer = imageResponse.data; 
+
+        return new Response(imageBuffer, {
+            status: 200, 
+            headers: {
+                'Content-Type': imageResponse.headers['content-type'], // Set the appropriate content type for the image
+                'Content-Length': String(imageBuffer.length),
+            }
+        });
+
+
+        
+    } catch (e: any) {
+        console.error(e);
+        return new Response(JSON.stringify({ error: e.message }), { status: 400 });
+    }
+}
+
+
+async function callAzureOpenAI(prompt: string) {
+    const endpoint = process.env.AZURE_OPENAI_ENDPOINT;
+    const key = process.env.AZURE_OPENAI_KEY;
+
+    const base_prompt = `Your job is to create festive avatars of North Pole characters, automatically rendering them in a cheerful, retro pixelated art style reminiscent of Super Nintendo 16-bit graphics. The GPT, with its concise and festive personality, focuses on the traditional holiday theme. It uses its creative judgment to fill in details, ensuring all avatars are in a pixelated style without requiring users to specify this. The goal is to provide users with delightful avatars that capture the essence of each character in a nostalgic, festive spirit.
```

Comment:
This description should be rewritten to match a direct dall-e call, i.e., don't talk about a GPT, don't talk about responding with a question. I'd also suggest putting this as a constant at file scope for maximum readability.

Reply:
Agreed - Maybe this goes in config.ts along with the other configuration constants?